### DISCUSSION TITLE
werkzeug.exceptions.BadRequest: 400 Bad Request: The browser (or proxy) sent a request that this server could not understand.

### DISCUSSION THREAD

**Discussion by pradeepdev-1995** [pallets/werkzeug#2525] [Q&A] [answered]
I tried the following code segment with flask

```server.py```

```
from flask import Flask,request
@app.route("/api_1/<query_param_1>/",methods=['GET', 'POST', 'PUT', 'DELETE'])
def myfunction(query_param_1):
 if request.method == 'POST':
     body = request.get_json()
     print("Body is "+ body)
```
Using flask, I could host the ```server.py``` file at localhost in a port address 7000 successfully.
Then I tried to call that API using from my ```client.py``` file

```client.py```
```
import requests
response = requests.post("http://0.0.0.0:7000/api_1/1235/")
```
Even though I am not passing any body to this function, I am expecting the printing ```Body is None```  in ```server.py``` console.
But it shows errors like this in the line - ```body = request.get_json()```
```  raise BadRequest()
werkzeug.exceptions.BadRequest: 400 Bad Request: The browser (or proxy) sent a request that this server could not understand.
```
Here are the package versions I am using

```
OS : Ubuntu 18.04
Python : 3.9.13
Werkzeug==2.1.1
requests==2.26.0
Flask==2.0.3
Flask-Cors==3.0.9
```

**The above code worked perfectly when I downgraded the Werkzeug version to 2.0.0**

**But I must need to work on the ```Werkzeug==2.1.1``` version. How to solve the issue?**

**Comment by ThiefMaster** [comment, marked as answer]
How do you expect to get a JSON body when your request does not contain anything JSON-related? :)

https://flask.palletsprojects.com/en/2.2.x/api/#flask.Request.get_json

Try `request.get_json(silent=True)` instead.

**Comment by pradeepdev-1995** [comment]
@ThiefMaster  Thank you. It worked
